python/src/equistore/operations/slice.py

```diff
@@ -0,0 +1,399 @@
+                raise ValueError(
+                    "Invalid sample name '"
+                    + name
+                    + "' - doesn't exist in input `TensorMap`."
+                )
+    if properties_to_slice is not None:
+        if not isinstance(properties_to_slice, Labels):
+            raise TypeError(
+                "properties_to_slice must be passed as a equistore `Labels` object."
+            )
+        property_names = tensormap.property_names
+        for name in properties_to_slice.names:
+            if name not in property_names:
+                raise ValueError(
+                    "Invalid property name '"
+                    + name
+                    + "' - doesn't exist in input `TensorMap`."
+                )
+
+    # Slice TensorMap
+    sliced_tensormap = _slice_tensormap(
+        tensormap,
+        samples_to_slice=samples_to_slice,
+        properties_to_slice=properties_to_slice,
+    )
+
+    # Calculate which blocks have been sliced to now be empty. True if any
+    # dimension of block.values is 0, False otherwise.
+    empty_blocks = np.array(
+        [np.any(np.array(block.values.shape) == 0) for _, block in sliced_tensormap]
+    )
+
+    # Issue warnings if some or all of the blocks are now empty.
+    if np.any(empty_blocks):
+        if np.all(empty_blocks):
+            warnings.warn(
+                "\nAll TensorBlocks in the sliced TensorMap are now empty, "
+                + "based on your choice of samples and/or properties to slice by. "
+            )
+        else:
+            warnings.warn(
+                "\nSome TensorBlocks in the sliced TensorMap are now empty, "
+                + "based on your choice of samples and/or properties to slice by. "
+                + "The keys of the empty TensorBlocks are:\n"
+                + str(tensormap.keys[empty_blocks])
+            )
+
+    return sliced_tensormap
+
+
+def slice_block(
+    tensorblock: TensorBlock,
+    samples_to_slice=None,
+    properties_to_slice=None,
+) -> TensorBlock:
+    """
+    Slices an input :py:class:`TensorBlock` along the samples and/or properties
+    dimension(s). ``samples_to_slice`` and ``properties_to_slice`` are
+    :py:class:`Labels` objects that specify the samples/properties
+    (respectively) names and indices that should be sliced, i.e. kept in the
+    output :py:class:`TensorBlock`.
+
+    Note that either ``samples_to_slice`` or ``properties_to_slice``, or both,
+    should be specified as input.
+
+    Example: take an input :py:class:`TensorBlock` of shape (100, 1, 6), where
+    there are 100 'samples', 1 'components', and 6 'properties'. Say we want to
+    slice this tensor along the samples and properties dimensions. As in the
+    code-block below, we can specify, for example, 4 samples and 3 properties
+    indices to keep. The returned :py:class:`TensorBlock` will have shape (4, 1,
+    3).
+
+    .. code-block:: python
+    
+        sliced_tensorblock = _slice_tensorblock(
```

Comment:
Why do refer to `_slice_tensorblock ` here instead of giving an example to the function you are documenting?

Reply:
Good point thank you!

Reply:
Change made in 1f8f8b8c81e91af663db59a154967c9c9707e186.